framework/src/main/java/org/checkerframework/framework/util/defaults/QualifierDefaults.java

```diff
@@ -1010,6 +1010,21 @@ public Void scan(@FindDistinct AnnotatedTypeMirror t, AnnotationMirror qual) {
             // Some defaults only apply to the top level type.
             boolean isTopLevelType = t == outer.type;
             switch (outer.location) {
+                case TYPE:
+                    if (outer.scope != null && outer.scope.getKind().isClass() && isTopLevelType) {
+                        AnnotationMirror annotation =
+                                outer.qualHierarchy.findAnnotationInHierarchy(
+                                        atypeFactory.elementCache.get(outer.scope).getAnnotations(),
+                                        qual);
+                        if (annotation == null
+                                || outer.qualHierarchy.isSubtypeQualifiersOnly(qual, annotation)) {
+                            outer.addAnnotation(t, qual);
+                            atypeFactory.elementCache.put(outer.scope, t);
+                        } else {
+                            // should report error;
```

Comment:
If the annotation supplied is incompatible with the one in `DefaultQualifier`, should we report error or we should keep the explicit qualifier? That is to say, in the following code, should we keep `@B` or should we report error?
```
@DefaultQualifier(
        value = A.class,
        locations = {TypeUseLocation.TYPE})
@B
public class DefaultQualifierTest {
    @B DefaultQualifierTest() {}
}
```